### PR TITLE
AUTO-437 Investigate lack of intensity data on hokuyo

### PR DESCRIPTION
**Pull request by richardw347**
This PR addresses an issue with the reliability of the intensity data publishing. The main issue is this line:

https://github.com/botsandus/urg_node2/blob/6632d1ce1859a2ad8ad11d1c1b1117178769f263/src/urg_node2.cpp#L297

The is_intensity_supported() function checks the device supports intensity data by trying to read a single stream of distance + intensity. There are main ways this function can fail, or even not perform the check (this especially occurs when the sensor reconnects). This PR adds a loop to retry this check a few times (limited by the error_limit parameter). 

I've tested this on the bench by disconnecting and reconnecting the LIDAR and it now recovers the intensity reliably (after a few retires in certain cases).

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -21,8 +19,12 @@
...
 
   <test_depend>ament_lint_auto</test_depend>
-  <test_depend>ament_lint_common</test_depend>
-  <test_depend>ament_cmake_gtest</test_depend>
+  <test_depend>ament_cmake_lint_cmake</test_depend>
+  <test_depend>ament_cmake_xmllint</test_depend>
+
+  <exec_depend>ament_lint_auto</exec_depend>
+  <exec_depend>ament_cmake_lint_cmake</exec_depend>
+  <exec_depend>ament_cmake_xmllint</exec_depend>
```
Comment (scheunemann) [inline package.xml]:
```suggestion
```

**Comment by richardw347** [comment]
I've updated this PR to include the same fix for multiecho mode and also reverted the disable of their tests and move that to a different [PR](https://github.com/botsandus/urg_node2/pull/3). This should allow us to pass on this and the other [bugfix](https://github.com/botsandus/urg_node2/pull/2) to the upstream repo.

**Comment by scheunemann** [comment]
Very cool @richardw347 thank you! I am getting this in the testing today and then let's get it in

**Comment by richardw347** [comment]
Just updated this with changes from master to resolve conflicts, looking forward to getting it in.

File: src/urg_node2.cpp [hunk 2]
```diff
@@ -1023,4 +1061,4 @@ rclcpp::Duration UrgNode2::get_angular_time_offset(void)
 }
 
 #include "rclcpp_components/register_node_macro.hpp"
-RCLCPP_COMPONENTS_REGISTER_NODE(urg_node2::UrgNode2)
+RCLCPP_COMPONENTS_REGISTER_NODE(urg_node2::UrgNode2)
```
Comment (scheunemann) [inline src/urg_node2.cpp]:
```suggestion
RCLCPP_COMPONENTS_REGISTER_NODE(urg_node2::UrgNode2)

```

:D

File: package.xml (hunk 1 above)
Comment (scheunemann) [reply package.xml]: Ups, this one got somehow stored and never sent. Maybe I forgot. Outdated anyway :+1:

File: src/urg_node2.cpp [hunk 3]
```diff
@@ -294,7 +294,23 @@ bool UrgNode2::connect()
...
+        RCLCPP_INFO(get_logger(), "Intensity mode init success");
+        use_intensity_ = true;
+        break;
+      }
+      RCLCPP_WARN(get_logger(), "Failed to init intensity mode");
+      error_count_ += 1;
+      if (error_count_ > error_limit_) {
+        error_count_ = 0;
+        use_intensity_ = false;
+        break
```
Comment (scheunemann) [inline src/urg_node2.cpp]:
```suggestion
        break;
```

Tested it and works for the robot (and compiles) :+1:

**Review by scheunemann — APPROVED** [review]

**Pull request merged**